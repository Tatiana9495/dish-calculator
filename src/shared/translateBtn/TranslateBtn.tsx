import React, { useState, useEffect } from 'react';

import styles from './index.module.scss';

interface ITranslateBtn {
  changeLanguage: (language: string) => void;
}

const data = [
  {
    id: 1,
    title: 'en',
  },
  {
    id: 2,
    title: 'ru',
  },
];

export const TranslateBtn: React.FC<ITranslateBtn> = (props: ITranslateBtn): JSX.Element => {
  const { changeLanguage } = props;
  const [lang, setLang] = useState('ru');

  const switchLanguage = (language: string) => {
    changeLanguage(language);
    setLang(language);
  };

  useEffect(() => {
    changeLanguage(lang);
  }, []);

  return (
    <div className={styles.container}>
      {data.map((item) => {
        return (
          <div className={styles.btnWrapper} onClick={() => switchLanguage(item.title)} key={`language-${item.id}`}>
            <div className={styles.circle}>{lang === item.title && <div className={styles.innerCircle}></div>}</div>
            <span>{item.title}</span>
          </div>
        );
      })}
    </div>
  );
};
