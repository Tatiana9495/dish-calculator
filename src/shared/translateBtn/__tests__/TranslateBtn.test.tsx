import React from 'react';
import ReactDOM from 'react-dom';
import { TranslateBtn } from '../TranslateBtn';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('renders correctly enzyme', () => {
  const wrapper = shallow(<TranslateBtn />)

  expect(toJson(wrapper)).toMatchSnapshot();
})