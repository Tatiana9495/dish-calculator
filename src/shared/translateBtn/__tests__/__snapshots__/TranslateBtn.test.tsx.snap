// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<div
  className="container"
>
  <div
    className="btnWrapper"
    key="language-1"
    onClick={[Function]}
  >
    <div
      className="circle"
    />
    <span>
      en
    </span>
  </div>
  <div
    className="btnWrapper"
    key="language-2"
    onClick={[Function]}
  >
    <div
      className="circle"
    >
      <div
        className="innerCircle"
      />
    </div>
    <span>
      ru
    </span>
  </div>
</div>
`;
